### PR TITLE
Call stopped node manager returns error

### PR DESCRIPTION
**Pull request by jjyr**
* Introduce `StopHandle` - a wrapper on AtomicBool
* The `start` function return error if node is already started.
* The following methods return error if node manager is stopped.
  * pay_invoice
  * list_nodes
  * send_to_address
  * estimate_tx_fee
  * sweep_wallet
  * create_address
  * create_lightning_invoice
  * get_balance
  * delete_all
  * bump_fee
* `estimate_fee_xxx` functions return `Result<u32,_>` instead of `u32`

### CODE REVIEW

File: mutiny-core/src/lib.rs [hunk 1]
```diff
@@ -1542,12 +1562,12 @@ impl<S: MutinyStorage> MutinyWallet<S> {
 
     /// Stops all of the nodes and background processes.
     /// Returns after node has been stopped.
-    pub async fn stop(&self) -> Result<(), MutinyError> {
+    pub async fn stop(&mut self) -> Result<(), MutinyError> {
         log_trace!(self.logger, "calling stop");
 
-        self.stop.store(true, Ordering::Relaxed);
```
Comment (EthanYuan) [inline mutiny-core/src/lib.rs:1548]: This line of code needs to seemingly not be removed.

Comment (jjyr) [reply mutiny-core/src/lib.rs:1548]: The stop field is removed becaused we use `Option` to store `node_manager`, the none value of  the `node_manager` field implies the node is stopped.

Comment (EthanYuan) [reply mutiny-core/src/lib.rs:1548]: It seems `MutinyLogger` child thread depends on this `stop` variable.

https://github.com/utxostack/mutiny-node/blob/develop/mutiny-core/src/lib.rs#L767

```rust
        let stop = Arc::new(AtomicBool::new(false));
        let logger = Arc::new(MutinyLogger::with_writer(
            stop.clone(),
            self.storage.clone(),
            self.session_id,
        ));
```

File: mutiny-wasm/src/lib.rs [hunk 2]
```diff
@@ -1379,7 +1380,7 @@ mod tests {
             .await
             .expect("failed to clear storage");
 
-        let nm = MutinyWallet::new(
+        let mut nm = MutinyWallet::new(
```
Comment (duanyytop) [inline mutiny-wasm/src/lib.rs:1383]: Is the `mut` required?

Comment (jjyr) [reply mutiny-wasm/src/lib.rs:1383]: stop function requires mut

File: mutiny-core/src/lib.rs (hunk 1 above)
Comment (jjyr) [reply mutiny-core/src/lib.rs:1548]: Fixed by adding an explicit stop function on the logger

**Review by EthanYuan — APPROVED** [review]

File: mutiny-core/src/lib.rs (hunk 1 above)
Comment (EthanYuan) [reply mutiny-core/src/lib.rs:1548]: Makes the end of background threads more controllable.

**Review by duanyytop — APPROVED** [review]

**Pull request merged**